libr/bin/format/elf/elf.c

```diff
@@ -75,6 +75,67 @@ static inline int UTX_MUL(ut64 *r, ut64 a, ut64 b) {
 
 #define round_up(a) ((((a) + (4) - (1)) / (4)) * (4))
 
+#define EF_MIPS_ABI_O32     0x00001000  /* O32 ABI.  */
+#define EF_MIPS_ABI_O64     0x00002000  /* O32 extended for 64 bit.  */
+#define EF_MIPS_ABI			0x0000f000
+#define mips_elf_check_machine(x) ((x)->e_machine == EM_MIPS)
+
+/*
+ * Return non-zero if HDR identifies an n64 ELF binary.
+ */
+#define elfn64_check_arch(hdr)                      \
```

Comment:
why all these macros? Aren't functions enough for what you need to do?

Reply:
Also, I think this macro syntax is only available on gcc but not everywhere (IIRC).

Reply:
Ok, I will modify it (I copy from linux kernel :) )